Automattic/jetpack
# Update/sync post meta add checks formeta args

Fixes #37691 

## Proposed changes:

* Adds defensive checks when filtering metadata.

### Other information:


## Does this pull request change what data or activity we track or use?


No
## Testing instructions:





Since I have not been able to reproduce the exact scenario in #37691 I just added defensive checks. 


The `filter_meta` method is used when `jetpack_sync_before_enqueue_added_post_meta`, `jetpack_sync_before_enqueue_updated_post_meta` and `jetpack_sync_before_enqueue_deleted_post_meta` are triggered while syncing so do the following to confirm there are no functional changes to the syncing of posts.
* In a JN site, add and delete some posts and ensure the meta is being synced.
